src/pages/Detail/DetailLoadManager.tsx

```diff
@@ -20,6 +20,11 @@ const DetailLoadManager: FC<RouteComponentProps<DetailsUrlProps>> = (props) => {
   const { dataset, model } = props.match.params;
 
   const auth = useAuth();
+  const authHeaders = new Headers();
+  if (auth.user?.access_token) {
+    authHeaders.append('Authorization', `Bearer ${auth.user!.access_token}`);
+  }
+  const fetchParams = { method: 'GET', headers: authHeaders };
   const store = useStore();
```

Comment:
(Applies to both blocks)

I'd prefer we use `useMemo` from React to get the headers, because this way we don't have to rerender the entire component if the auth changes. Two steps to do this:

1) We use React's `useMemo` hook to get the headers back.

```typescript
const fetchParams = useMemo(() => {
  const authHeaders = new Headers();
  if (auth.user?.access_token) {
    authHeaders.append('Authorization', `Bearer ${auth.user!.access_token}`);
  }
  return {method: 'GET', headers: authHeaders };
}, [auth.user?.access_token]);
```

Note that we have to use React's `useMemo` instead of the mobx local observable, as useAuth() is not integrated with Mobx (which is fine).

2) Update the `useEffect` dependency array to also include `fetchParams`:

```typescript
useEffect(() => {
  // keep code as you had it
}, [state.loadState, fetchParams]);
```

(that's just the dependency array in this instance, you'll just want to add `fetchParams` to each instance)

Reply:
There's a dependency I have disabled, `react-hooks/exhaustive-deps`, because of mobx. That rule makes sure that you include all dependencies in your dependency array. However, disabling this check means that we can sometimes miss improper configurations for the dependency array when they are React-specific.